app/screens/get-started-screen/device-account-modal.tsx

```diff
@@ -42,38 +39,82 @@ export const DeviceAccountModal: React.FC<DeviceAccountModalProps> = ({
   appCheckToken,
 }) => {
   const { setAuthenticatedWithDeviceAccount } = useAppConfig()
+  const {
+    appConfig: {
+      galoyInstance: { restUrl },
+    },
+  } = useAppConfig()
 
-  const [userDeviceLogin, { loading }] = useUserLoginDeviceMutation()
   const [hasError, setHasError] = React.useState(false)
+  const [loading, setLoading] = React.useState(false)
+
   const { LL } = useI18nContext()
   const navigation =
     useNavigation<StackNavigationProp<RootStackParamList, "getStarted">>()
 
   const createDeviceAccountAndLogin = async () => {
     try {
+      setLoading(true)
+      const credentials = await Keychain.getInternetCredentials("device-account")
+
+      let username: string
+      let password: string
+
+      if (credentials) {
+        username = credentials.username
+        password = credentials.password
+      } else {
+        username = await generateSecureRandomUUID()
+        password = await generateSecureRandomUUID()
+        const setPasswordResult = await Keychain.setInternetCredentials(
+          "device-account",
+          username,
+          password,
+        )
+        if (!setPasswordResult) {
+          throw new Error("Error setting device account credentials")
+        }
+      }
+
       logAttemptCreateDeviceAccount()
-      const { data } = await userDeviceLogin({
-        variables: {
-          input: {
-            jwt: appCheckToken,
-          },
+
+      const auth = Buffer.from(`${username}:${password}`, "utf8").toString("base64")
+
+      const res = await fetch(restUrl + "/auth/create/device-account", {
+        method: "POST",
+        headers: {
+          "Authorization": `Basic ${auth}`,
+          "Appcheck": appCheckToken,
+          "Content-Type": "application/json",
         },
+        body: JSON.stringify({
+          deviceId: await generateSecureRandomUUID(), // TODO add deviceId
+        }),
       })
 
-      const sessionToken = data?.userLoginDevice.authToken
-      if (sessionToken) {
-        logCreatedDeviceAccount()
-        setAuthenticatedWithDeviceAccount()
-        navigation.replace("Primary")
-        closeModal()
-        return
+      const data: {
+        result: string | undefined
+      } = await res.json()
+
+      setLoading(false)
+
+      const sessionToken = data.result
+
+      if (!sessionToken) {
+        throw new Error("Error getting session token")
       }
+
+      logCreatedDeviceAccount()
+      analytics().logLogin({ method: "device" })
+      setAuthenticatedWithDeviceAccount(sessionToken)
+      navigation.replace("Primary")
+      closeModal()
     } catch (error) {
+      setHasError(true)
+      setLoading(false)
       logCreateDeviceAccountFailure()
-      console.log("Error creating device account: ", error)
+      console.log("Error with device account: ", error)
```

Comment:
maybe we could/should have an Alert here instead of a console.log? not sure what kind of error we can expect. I guess the main question I have is on whether all android device works with the react-native-keychain package.

Reply:
In the case of an error it will show an error modal, so I'm not sure if it is valuable to alert the user with. In order to detect the error, we could log it with crashlytics.